android/src/main/java/com/wix/interactable/TouchBlocker.java

```diff
@@ -12,19 +12,28 @@
 
 public class TouchBlocker extends ViewGroup {
     public final static String TAG = "TouchBlocker";
+    private float y1, y2, x1, x2;
```

Comment:
No need save all these fields, enough initiate one field `PointF touchStartLocation` on ACTION_DOWN event, and on ACTION_MOVE to do :
```
            case MotionEvent.ACTION_MOVE:
                if (blockVerticalSwipe) {
                    float deltaY = ev.getY() - touchStartLocation.y;
                    if (Math.abs(deltaY) > mTouchSlop) {
                        getParent().requestDisallowInterceptTouchEvent(true);
                    }
                    break;
                }
                if (blockHorizontalSwipe) {
                    float deltaX = ev.getX() - touchStartLocation.x;
                    if (Math.abs(deltaX) > mTouchSlop) {
                        getParent().requestDisallowInterceptTouchEvent(true);
                    }
                    break;
                }
```